src/rdkafka_ssl.c

```diff
@@ -544,7 +544,7 @@ static int rd_kafka_transport_ssl_verify (rd_kafka_transport_t *rktrans) {
                 return -1;
         }
 
-        if ((rl = SSL_get_verify_result(rktrans->rktrans_ssl)) != X509_V_OK) {
+        if (rktrans->rktrans_rkb->rkb_rk->rk_conf.ssl.enable_verify && (rl = SSL_get_verify_result(rktrans->rktrans_ssl)) != X509_V_OK) {
```

Comment:
I suggest simplifying the logic by adding the enable_verify check to the beginning of this function, and if set simply return 0.

Reply:
> and if set simply return 0  

opposite  **is not set** (set to false)